enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,240 @@
+- OpenShift Ingress Operator creates a watcher and reconciles whenever changes are observed
+  in the aforementioned ConfigMap objects.
+- Openshift Router uses the operator created ConfigMap as the default destination CA certificate
+  to verify services' certificate, when a route with re-encrypt termination type is created
+  without destinationCA configured.
+- OpenShift router watches and reloads whenever CA certificate bundle changes.
+
+![alt Workflow diagram](admin-configured-ca-bundle.png)
+
+#### Variation [optional]
+None
+
+### API Extensions
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+An OpenShift administrator will have to create a ConfigMap with the fixed name `admin-ca-bundle`
+and with the list of all trusted CA certificates in `openshift-config` namespace.
+
+```yaml
+apiVersion: v1
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    [...]
+    -----END CERTIFICATE-----
+kind: ConfigMap
+metadata:
+  name: admin-ca-bundle
+  namespace: openshift-config
+```
+
+OpenShift Ingress operator during reconciliation would look for ConfigMaps bearing the names
+`admin-ca-bundle` and `service-ca-bundle` in `openshift-config` and`openshift-ingress`
+namespaces respectively and create a ConfigMap with the name `ingress-ca-bundle` in
+`openshift-ingress` namespace.
+
+```yaml
+apiVersion: v1
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    <service-ca>
+    -----END CERTIFICATE-----
+    -----BEGIN CERTIFICATE-----
+    <admin-ca>
+    -----END CERTIFICATE-----    
+kind: ConfigMap
+metadata:
+  name: ingress-ca-bundle
+  namespace: openshift-ingress
+```
+
+OpenShift Ingress operator should create a watcher to observe `admin-ca-bundle` and
+`service-ca-bundle` for any changes and update `ingress-ca-bundle` accordingly.
+
+OpenShift Router currently uses the CA certificate provided as a process argument
+`default-destination-ca-path` or set as an environment variable `DEFAULT_DESTINATION_CA_PATH`
+as the CA certificate for verifying the service's certificate for re-encrypt termination type.
+
+CA certificate bundle is currently made available to the router using the volume mount option 
+at the path `/var/run/configmaps/service-ca/service-ca.crt` which will be updated to 
+`/var/run/configmaps/ca-trust/ca-bundle.crt` with the `ingress-ca-bundle` content.
+
+OpenShift Router should create a file watcher to observe `/var/run/configmaps/ca-trust/ca-bundle.crt`
+file and reload when the file content has been modified.
+
+### Risks and Mitigations
+
+N/A.
+
+### Drawbacks
+
+When `ingress-ca-bundle` ConfigMap contents made available to OpenShift Router as a file
+though volume mount is updated, certain [delay](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically) 
+is expected for the new changes to be propagated to the pod.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A.
```

Comment:
Please add that the solution is only for a non-sharded environment, in which there is only a default Ingress Controller.

Reply:
Non-Goals section updated to indicate the `ingress-ca-bundle` is common for all IngressControllers.